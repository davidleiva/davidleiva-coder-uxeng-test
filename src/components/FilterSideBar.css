.FilterSideBar {
  width: 224px;
  padding: 16px;
  background-color: rgb(4, 6, 11);
  color: rgba(255, 255, 255, 0.87);
}

.FilterSideBar__Title.MuiTypography-h6 {
  color: hsl(218, 32%, 77%);
  font-size: 12px;
  font-weight: 600;
  letter-spacing: 0.1em;
  text-transform: uppercase;
  margin-bottom: 0;
  margin-left: 8px;
}

.FilterSideBar .FilterSideBar__Header svg.MuiSvgIcon-root path {
  fill: hsl(218, 32%, 77%);
}

.FilterSideBar .MuiAccordion-root.MuiPaper-root {
  background-color: rgb(4, 6, 11);
  border-top-left-radius: 0;
  border-top-right-radius: 0;
  border-bottom-left-radius: 0;
  border-bottom-right-radius: 0;
  box-shadow: none;
  background-image: none;
  color: hsl(218, 32%, 77%);
}

.FilterSideBar .MuiAccordion-root.MuiPaper-root svg path {
  fill: hsl(218, 32%, 77%);
}

.FilterSideBar .MuiAccordion-root.MuiPaper-root:hover,
.FilterSideBar .MuiAccordion-root.MuiPaper-root.Mui-expanded {
  color: hsl(220, 55%, 98%);
}

.FilterSideBar .MuiAccordion-root.MuiPaper-root:hover svg path {
  color: hsl(220, 55%, 98%);
}

.FilterSideBar .MuiAccordionDetails-root {
  padding: 0;
}

.FilterSideBar .MuiAccordion-root.MuiPaper-root .MuiAccordionSummary-root {
  padding-left: 0;
  padding-right: 0;
  height: 36px;
}

.FilterSideBar .MuiAccordionSummary-content>.MuiTypography-root,
.FilterSideBar .MuiListItemText-root>.MuiTypography-root {
  font-size: 12px;
}

.FilterSideBar .MuiAccordion-root.MuiPaper-root:before {
  content: none;
}

.FilterSideBar .MuiAccordion-root.MuiPaper-root .MuiSvgIcon-root {
  width: 0.66em;
  height: 0.66em;
}

.FilterSideBar .MuiAccordionSummary-content.Mui-expanded {
  margin: 4px 0;
}

.FilterSideBar .MuiAccordion-root .MuiButtonBase-root {
  min-height: initial;
  height: 36px;
}

.FilterSideBar .MuiAccordion-root.MuiPaper-root.Mui-expanded {
  margin-bottom: 0;
  margin-top: 0;
}

.FilterSideBar .MuiListItemButton-root.justifyEnd .MuiListItemIcon-root {
  justify-content: end;
}

.FilterBar__Divider.MuiDivider-root {
  margin: 8px auto;
}

.FilterSearchUser input,
.FilterSearchUser label,
.FilterSearchUser .MuiAvatar-circular {
  font-size: 12px;
}

.MuiListItemButton-root .MuiListItemIcon-root,
.MuiListItemButton-root .MuiAvatar-root {
  min-width: 0;
}

.MuiListItemButton-root .MuiAvatar-root {
  margin-right: 8px;
}

.FilterSideBar .FilterSideBar__Link {
  color: rgba(255, 255, 255, 0.87);
  font-size: 12px;
  display: flex;
  align-items: center;

}

.FilterSideBar .FilterSideBar__Link:hover {
  color: rgba(255, 255, 255, 1);
  font-size: 12px;
}

.FilterSideBar .FilterSideBar__Link svg {
  width: 0.5em;
  height: 0.5em;
  margin-right: 8px;
}


/* TODO: This is only a Fix for storybook */
.QueryEditor .css-12wnr2w-MuiButtonBase-root-MuiCheckbox-root {
  color: rgba(255, 255, 255, 0.7);
}