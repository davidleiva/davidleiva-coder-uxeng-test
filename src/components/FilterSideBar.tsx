import { useState } from 'react';
import {FilterAccordion} from './FilterAccordion';
import Typography from "@mui/material/Typography";
import FilterListIcon from '@mui/icons-material/FilterList';
import { QueryEditor } from './QueryEditor';
import { PREDEFINED_FILTERS } from '../constants';
import Divider from "@mui/material/Divider";
import { FilterStatesContext } from '../ContextManagement';

const FilterSideBar = () => {
    
    const setFiltersState = (filters:any) => {
        setState({...filterState, filters: filters})
    }

    const initState = {
        filters: [],
        setFiltersState: setFiltersState
    } 
    
    const [filterState, setState] = useState({...initState})

    const [activeAccordion, setActiveAccordion] = useState(0);

    const handleActiveAccordion = (value:number, params:object) => {
        value !== activeAccordion ? setActiveAccordion(value) : setActiveAccordion(NaN),
        setFiltersState({...params});
    };



    return (
        <FilterStatesContext.Provider value={initState}>
        <div className="FilterSideBar">
            <div
                className="FilterSideBar__icon"
                style={{
                    display: 'flex',
                    alignItems: 'center',
                    marginBottom: '8px'
                }}
            >
                <FilterListIcon />
                <Typography variant="h6" gutterBottom className="FilterSideBar__header">
                    Filter
                </Typography>
            </div>
            <FilterAccordion
                title={'Predefined filters'} 
                filters={PREDEFINED_FILTERS}
                defaultExp={true}
                isExpanded={activeAccordion === 0}
                handleChange={(params:object) => handleActiveAccordion(0,params)}
                filterState={filterState}
            />
            <FilterAccordion
                title={'Resource type'} 
                filters={PREDEFINED_FILTERS}
                isExpanded={activeAccordion === 1}
                handleChange={(params:object) => handleActiveAccordion(1,params)}
            />
            <FilterAccordion
                title={'Actions'} 
                filters={PREDEFINED_FILTERS}
                isExpanded={activeAccordion === 2}
                handleChange={(params:object) => handleActiveAccordion(2,params)}
            />
            {/* <FilterAccordion
                title={'Users'} 
                filters={<FilterSBUsers />}
                isExpanded={activeAccordion === 3}
                handleChange={(params:object) => handleActiveAccordion(3,params)}
            /> */}
            <Divider className="FilterBar__Divider"/>
            <QueryEditor />
        </div>
        </FilterStatesContext.Provider>
      )
};

export default FilterSideBar;