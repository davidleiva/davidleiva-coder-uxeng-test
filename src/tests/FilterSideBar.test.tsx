import {describe, expect, test} from 'vitest';
import FilterSideBar from '../components/FilterSideBar';

describe("Accordion test without props", () => {
    test("FilterSideBar renders ok", () => {
        
        const fc = <FilterSideBar />;

        expect(fc).toMatchSnapshot()
    })
})