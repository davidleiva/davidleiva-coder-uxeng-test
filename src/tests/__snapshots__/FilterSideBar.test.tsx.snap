// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Accordion test > FilterSideBar renders ok 1`] = `<FilterSideBar />`;
